Flaky test: Test/UpdateAddresses_RetryFromFirstAddr

https://github.com/grpc/grpc-go/pull/4718/checks?check_run_id=3468320856

```
--- FAIL: Test (22.16s)
    --- FAIL: Test/UpdateAddresses_RetryFromFirstAddr (5.01s)
        tlogger.go:101: clientconn.go:251 [core] parsed scheme: "whatever"  (t=+1.374424ms)
        tlogger.go:101: resolver_conn_wrapper.go:100 [core] ccResolverWrapper: sending update to cc: {[{127.0.0.1:37189  <nil> 0 <nil>} {127.0.0.1:35219  <nil> 0 <nil>} {127.0.0.1:46861  <nil> 0 <nil>}] <nil> <nil>}  (t=+1.635329ms)
        tlogger.go:101: clientconn.go:1141 [core] Subchannel Connectivity change to CONNECTING  (t=+1.945935ms)
        tlogger.go:101: clientconn.go:1252 [core] Subchannel picks a new address "127.0.0.1:37189" to connect  (t=+2.058937ms)
        tlogger.go:101: clientconn.go:435 [core] Channel Connectivity change to CONNECTING  (t=+4.089474ms)
        tlogger.go:101: clientconn.go:1141 [core] Subchannel Connectivity change to READY  (t=+4.514481ms)
        tlogger.go:101: clientconn.go:435 [core] Channel Connectivity change to READY  (t=+4.925489ms)
        tlogger.go:101: clientconn.go:1141 [core] Subchannel Connectivity change to IDLE  (t=+5.102392ms)
        tlogger.go:101: clientconn.go:435 [core] Channel Connectivity change to IDLE  (t=+5.346996ms)
        clientconn_test.go:1008: timed out waiting for server1 to be contacted
        tlogger.go:101: clientconn.go:435 [core] Channel Connectivity change to SHUTDOWN  (t=+5.003964007s)
        tlogger.go:101: clientconn.go:1141 [core] Subchannel Connectivity change to SHUTDOWN  (t=+5.00412731s)
        clientconn_test.go:962: accept tcp 127.0.0.1:35219: use of closed network connection
        clientconn_test.go:944: accept tcp 127.0.0.1:37189: use of closed network connection
FAIL
FAIL	google.golang.org/grpc	40.914s
```

--- a/clientconn_test.go
+++ b/clientconn_test.go
@@ -1067,8 +1067,14 @@ func stayConnected(cc *ClientConn) {
 	defer cancel()
 
 	for {
-		cc.Connect()
-		if state := cc.GetState(); state == connectivity.Shutdown || !cc.WaitForStateChange(ctx, state) {
+		state := cc.GetState()
+		switch state {
+		case connectivity.Idle:
+			cc.Connect()
+		case connectivity.Shutdown:
+			return
+		}
+		if !cc.WaitForStateChange(ctx, state) {
 			return
 		}
 	}

